pkg/agent/agent.go

```diff
@@ -628,6 +687,74 @@ func annotatePod(k8sClient kubernetes.Interface, req *cni.CNIRequest, configData
 	return nil
 }
 
+func (s *NimbessAgent) processNotification() (bool) {
+
+	fibRequest := <-s.notifications
+		log.Debugf("Processing FIB Request %s MAC %s", fibRequest.Command, fibRequest.MAC)
+
+	s.Mu.Lock()
+	defer s.Mu.Unlock()
+
+	// pipeline's fib points to its parent metapipeline fib
+	pipeline, ok := s.Pipelines[fibRequest.Port]
+
+	if (!ok) {
+		log.Errorf("Failed to find metaPipeline and its fib")
```

Comment:
are you missing a return here if cannot find metapipeline/fib? Otherwise I get:

time="2019-10-17T13:14:14Z" level=debug msg="Processing FIB Request LEARN MAC 52:71:42:3c:f9:01" func="github.com/nimbess/nimbess-agent/pkg/agent.(*NimbessAgent).processNotification" file="/go/src/github.com/nimbess/nimbess-agent/pkg/agent/agent.go:693"
time="2019-10-17T13:14:14Z" level=error msg="Failed to find metaPipeline and its fib" func="github.com/nimbess/nimbess-agent/pkg/agent.(*NimbessAgent).processNotification" file="/go/src/github.com/nimbess/nimbess-agent/pkg/agent/agent.go:702"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x13def10]

goroutine 27 [running]:
github.com/nimbess/nimbess-agent/pkg/agent.(*NimbessAgent).processNotification(0xc00047c200, 0x0)
	/go/src/github.com/nimbess/nimbess-agent/pkg/agent/agent.go:707 +0x5b0
github.com/nimbess/nimbess-agent/pkg/agent.(*NimbessAgent).runNotifications(0xc00047c200)
	/go/src/github.com/nimbess/nimbess-agent/pkg/agent/agent.go:752 +0x59
created by github.com/nimbess/nimbess-agent/pkg/agent.(*NimbessAgent).Run
	/go/src/github.com/nimbess/nimbess-agent/pkg/agent/agent.go:815 +0x17c